notification_controller/controller/controller_test.go

```diff
@@ -115,6 +115,7 @@ func TestInit(t *testing.T) {
 		dynamicClient,
 		nil,
 		"default",
+		[]string{},
```

Comment:
Would you mind adding some unit tests to cover the apps-in-any-namespace behavior? I think at the minimum we should make sure the notifications controller refuses to manage apps outside the specified namespaces.